File: app/src/main/java/harkor/mycryptocurrency/view/DialogSettings.java
package harkor.mycryptocurrency.view;

import android.app.AlertDialog;
import android.app.Dialog;
import android.app.DialogFragment;
import android.content.Context;
import android.content.DialogInterface;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.RadioButton;

import harkor.mycryptocurrency.Interfaces.SettingsDialogInterface;
import harkor.mycryptocurrency.R;
import harkor.mycryptocurrency.viewmodel.SettingsDialogViewmodel;

public class DialogSettings extends DialogFragment implements SettingsDialogInterface{
    RadioButton radioBtc;
    RadioButton radioUsd;
    RadioButton radioEur;
    RadioButton radioPln;
    View mView;

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        AlertDialog.Builder builder=new AlertDialog.Builder(getActivity());
        mView=getActivity().getLayoutInflater().inflate(R.layout.settings_dialog,null);
        radioBtc=mView.findViewById(R.id.radio_btc);
        radioUsd=mView.findViewById(R.id.radio_usd);
        radioEur=mView.findViewById(R.id.radio_eur);
        radioPln=mView.findViewById(R.id.radio_pln);
        final SettingsDialogViewmodel settingsDialogViewmodel=new SettingsDialogViewmodel(this);
        settingsDialogViewmodel.initially();
        builder.setView(mView)
                .setNegativeButton(R.string.cancel, new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialogInterface, int i) {
                    }
                })
                .setPositiveButton(R.string.ok, new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialogInterface, int i) {
                        settingsDialogViewmodel.okClick();
                    }
                });
        return builder.create();
    }

    @Override
    public int getCurrencyId() {
        final SharedPreferences sharedPref =mView.getContext().getSharedPreferences("harkor.myCryptocurrency", Context.MODE_PRIVATE);
        return sharedPref.getInt("currencyCode",1);
    }

    @Override
    public void checkDefault(int checked) {
        switch (checked){
            case 1: radioUsd.setChecked(true); break;
            case 2: radioEur.setChecked(true); break;
            case 3: radioPln.setChecked(true); break;
            case 4: radioBtc.setChecked(true); break;
        }
    }

    @Override
    public int whatIsChecked() {
        if(radioUsd.isChecked()){
            return 1;
        }else if(radioEur.isChecked()){
            return 2;
        }else if(radioPln.isChecked()){
            return 3;
        }else{
            return 4;
        }
    }

    @Override
    public void saveShared(int currencyId) {
        final SharedPreferences sharedPref =mView.getContext().getSharedPreferences("harkor.myCryptocurrency", Context.MODE_PRIVATE);
        SharedPreferences.Editor editor=sharedPref.edit();
        editor.putInt("currencyCode",currencyId);
        editor.apply();
    }
}

File: app/src/main/java/harkor/mycryptocurrency/view/DialogSettings.kt
package harkor.mycryptocurrency.view

import android.app.AlertDialog
import android.app.Dialog
import android.os.Bundle
import androidx.fragment.app.DialogFragment
import harkor.mycryptocurrency.services.SharedPref
import android.content.Intent
import android.net.Uri
import harkor.mycryptocurrency.databinding.SettingsDialogBinding


class DialogSettings(val notifyCurrencyChange: NotifyCurrencyChange) : DialogFragment() {
    private lateinit var binding: SettingsDialogBinding

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        return activity?.let {
            val builder = AlertDialog.Builder(it)
            binding = SettingsDialogBinding.inflate(layoutInflater)
            val currencyId = SharedPref.getDefaultCurrency(binding.root.context)
            when (currencyId) {
                1 -> binding.radioUsd.isChecked = true
                2 -> binding.radioEur.isChecked = true
                3 -> binding.radioPln.isChecked = true
                4 -> binding.radioBtc.isChecked = true
            }
            binding.settingsPrivacyPolicy.setOnClickListener{
                val browserIntent = Intent(Intent.ACTION_VIEW, Uri.parse("https://my-cryptocurrency.flycricket.io/privacy.html"))
                startActivity(browserIntent)
            }
            binding.settingsCancel.setOnClickListener {
                dialog?.cancel()
            }

            binding.settingsOk.setOnClickListener {
                val newCurrency:Int = when {
                    binding.radioUsd.isChecked -> 1
                    binding.radioEur.isChecked -> 2
                    binding.radioPln.isChecked -> 3
                    else -> 4
                }
                if(newCurrency!=currencyId){
                    SharedPref.saveDefaultCurrency(newCurrency,binding.root.context)
                    notifyCurrencyChange.change()
                }
                dialog?.cancel()
            }

            builder.setView(binding.root)
            builder.create()
        } ?: throw IllegalAccessException("Activity cannot be null")
    }
}
